smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceOperationInputOutputValidator.java

```diff
@@ -0,0 +1,309 @@
+                                      memberShape.getMemberName()))));
+            }
+        });
+    }
+
+    private void validateConflictingProperties(
+        List<ValidationEvent> events,
+        StructureShape shape,
+        Map<String, Set<MemberShape>> propertyToMemberMappings
+    ) {
+        for (Map.Entry<String, Set<MemberShape>> entry : propertyToMemberMappings.entrySet()) {
+            if (entry.getValue().size() > 1) {
+                events.add(error(shape, String.format(
+                        "This shape contains members with conflicting property names that resolve to '%s': %s",
+                        entry.getKey(),
+                        entry.getValue().stream().map(MemberShape::getMemberName)
+                            .collect(Collectors.joining(", ")))));
+            }
+        }
+    }
+
+    private void validateMember(
+        List<ValidationEvent> events,
+        String lifecycleOperationName,
+        Model model,
+        ResourceShape resource,
+        MemberShape member,
+        Set<String> identifierMembers,
+        Map<String, ShapeId> properties,
+        Map<String, Set<MemberShape>> propertyToMemberMappings
+    ) {
+        String propertyName = getPropertyNameFromMember(member);
+        propertyToMemberMappings.computeIfAbsent(propertyName,
+            m -> new TreeSet<>()).add(member);
+
+        if (properties.containsKey(propertyName)) {
+            if (!properties.get(propertyName).equals(member.getTarget())) {
+                events.add(error(resource, String.format("The resource property `%s` has a conflicting"
+                    + " target shape `%s` with the `%s` lifecycle operation which targets `%s`.",
+                    propertyName, lifecycleOperationName, properties)));
+            }
+        } else {
+            events.add(error(member, String.format("Member `%s` targets does not target a resource property",
+                member.getMemberName(),
+                propertyName)));
+        }
+    }
+
+    private Optional<StructureShape> getOutputPropertiesShape(
+        Model model,
+        OperationIndex operationIndex,
+        OperationShape operation
+    ) {
+        return operationIndex.getOutputMembers(operation).values().stream()
+            .filter(member -> member.hasTrait(NestedPropertiesTrait.class))
+            .map(member -> model.getShape(member.getTarget()).flatMap(s -> s.asStructureShape()))
+            .findFirst().orElse(operationIndex.getOutputShape(operation).flatMap(s -> s.asStructureShape()));
+    }
+
+    private Optional<StructureShape> getInputPropertiesShape(
+        Model model,
+        OperationIndex operationIndex,
+        OperationShape operation
+    ) {
+        return operationIndex.getInputMembers(operation).values().stream()
+            .filter(member -> member.hasTrait(NestedPropertiesTrait.class))
+            .map(member -> model.getShape(member.getTarget()).flatMap(s -> s.asStructureShape()))
```

Comment:
If you add to the TargetValidator, then you could just use model.expectShape(x, StructureShape.class). See https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/TargetValidator.java#L128

Reply:
Resource property target shape validation could see a mismatch with one or more of an operation's input/output structure members, and neither is necessarily the correct target. The valid target for a resource property is dependent on a given operation, and the input or output structure members of that operations map to that property.